### ISSUE TITLE
no-reference with type definition files

### ISSUE THREAD

**Issue by DataTables** [palantir/tslint#1562]
### Bug Report
- **TSLint version**: `3.15.1`
- **TypeScript version**: `2.0.0 beta`
- **Running TSLint via**: Sublime Linter
#### TypeScript code being linted

``` ts
/// <reference path="../types/node.d.ts" />
```

with `tslint.json` configuration:

``` json
{
  "no-reference": true
}
```
#### Actual behavior

Reference not allowed warning.
#### Expected behavior

Because this is a type definition file my understanding is that a `reference` is [still the way to import it](https://github.com/Microsoft/TypeScript/issues/2242#issuecomment-836941810 rather than with ES6 modules. As a result the `<reference` is valid in this use case with no ES6 alternative.

I wonder if it would be appropriate to allow the rule to allow type definition files?

**Comment by JKillian** [comment]
@DataTables better in this situation to just add your typings files to your list of files via your `tsconfig.json` file. Then you don't have to include comments like that anywhere in your source files.

Perhaps that should be better explained in the documentation though?

**Comment by DataTables** [comment]
Including that in the documentation sounds good to me - thanks!
